src/super_context.ml

```diff
@@ -772,23 +801,36 @@ module Action = struct
     (t, acc)
 
   let expand_step2 ~dir ~dynamic_expansions ~deps_written_by_user ~map_exe t =
-    U.Partial.expand t ~dir ~map_exe ~f:(fun var _syntax_version ->
+    U.Partial.expand t ~dir ~map_exe ~f:(fun var syntax_version ->
       let key = String_with_vars.Var.full_name var in
       let loc = String_with_vars.Var.loc var in
       match String.Map.find dynamic_expansions key with
       | Some _ as opt -> opt
       | None ->
+        let first_dep () =
+          Some (
+            match deps_written_by_user with
+            | [] ->
+              Loc.warn loc "Variable '%s' used with no explicit \
+                            dependencies@." key;
+              [Value.String ""]
+            | v :: _  -> [Path v]
+          )
+        in
         match key with
         | "<" ->
-          Some
-            (match deps_written_by_user with
-             | [] ->
-               Loc.warn loc "Variable '<' used with no explicit \
-                             dependencies@.";
-               [Value.String ""]
-             | dep :: _ ->
-               [Path dep])
-        | "^" -> Some (Value.L.paths deps_written_by_user)
+          if syntax_version < (1, 0) then
+            first_dep ()
+          else
+            Loc.fail loc "Variable '<' is renamed to 'first-dep' in dune"
```

Comment:
The error messages are not consistent: some say `in dune files`, some say `in dune` and some say nothing. `in dune files` seems best to me

Reply:
Agreed. I made all the errors a little more consistent. Full consistency will probably have to wait for the refactor that makes variable name changes first class.